lib/models/device.coffee

```diff
@@ -249,6 +249,41 @@ exports.getApplicationName = (uuid, callback) ->
 	exports.get(uuid).get('application_name').nodeify(callback)
 
 ###*
+# @summary Get application container information
+# @name getApplicationInfo
+# @public
+# @function
+# @memberof resin.models.device
+#
+# @param {String} uuid - device uuid
+# @fulfil {Object} - application info
+# @returns {Promise}
+#
+# @example
+# resin.models.device.getApplicationInfo('7cf02a6').then(function(appInfo) {
+# 	console.log(appInfo);
+# });
+#
+# @example
+# resin.models.device.getApplicationInfo('7cf02a6', function(error, appInfo) {
+# 	if (error) throw error;
+# 	console.log(appInfo);
+# });
+###
+exports.getApplicationInfo = (uuid, callback) ->
+	exports.get(uuid).then (device) ->
```

Comment:
The way I see it, the supervisor API proxy is a transparent intermediate and therefore it's the SDK's responsibility to be updated with the supervisor API.

I also don't think there is a dichotomy here, we can have the SDKs track supervisor API versions *and* have a robust/scalable way to report deprecations.